### DISCUSSION TITLE
评论时提示Failed to fetch

### DISCUSSION THREAD

**Discussion by sikonn** [walinejs/waline#996] [Q&A] [answered]
评论时提示Failed to fetch怎么办，已经删了重开了一次还是不行，求救
这是我的博客网址https://changingmoments.vercel.app/

**Comment by sikonn** [comment]
还有一个问题，HTML 引入的时候应该放在哪里，放在页脚里总觉得怪怪的！

**Reply by lizheming** [reply]
JS 放哪里问题不大，你把插入评论的位置放对就可以了

**Reply by sikonn** [reply]
那插入评论应该放在哪个位置呢

**Reply by lizheming** [reply]
你觉得放在哪里合适就放在哪里

**Reply by sikonn** [reply]
要是想放在文章的后面，而不是页脚的话应该放在哪里比较好？

**Reply by lizheming** [reply]
看了下你的主题，放在 `<div class="main-inner"></div>` 这个元素里最后就可以了。

**Reply by sikonn** [reply]
呜呜，还有一个问题。我想改成暗黑模式也能用的那种，auto和[]两种方法都试了一下，不行，改回来发现我的暗黑模式没有了，能不能帮我看一下?

**Reply by sikonn** [reply]
![屏幕截图(256)](https://user-images.githubusercontent.com/104444438/166696343-7576baa6-2b2b-4c50-ab57-b716488f18f0.png)
![屏幕截图(257)](https://user-images.githubusercontent.com/104444438/166696363-26501a6b-909d-4e06-97fe-3ca538ba3fda.png)
我是改了这两个，希望能帮我看一下！谢谢！

**Reply by HJCheng0602** [reply]
还有一种可能：
把vercel的[Vercel Authentication]设置关掉，我找了半天然后发现我开了这个，一关就好了

---
There is another possibility:
Turn off the [Vercel Authentication] setting of vercel, I searched for a long time and found that I turned on this, and it would be fine for one level

**Comment by lizheming** [comment, marked as answer]
博客里我没有看到有插入 Waline 评论的地方，无法帮忙查看

**Reply by sikonn** [reply]
不好意思刚刚想改一下删去了一会，现在应该可以看到了！！！https://changingmoments.vercel.app/

**Reply by lizheming** [reply]
没有改 serverURL ....
![image](https://user-images.githubusercontent.com/424491/166660576-c007d23a-8de5-44c6-b1c9-d0bb92d7faab.png)

**Reply by sikonn** [reply]
我改了发现还是不行！

**Reply by sikonn** [reply]
Unexpected end of JSON input报错变成了这个！

**Reply by lizheming** [reply]
https://waline.js.org/guide/get-started.html 请对着快速上手操作一遍，不要想当然。serverURL 是指评论服务的地址，不是你的博客地址！

**Reply by sikonn** [reply]
真的可以了呜呜呜！谢谢你！

**Reply by Chenm4** [reply]
请问一下这个serverURL评论服务的地址指的是哪个？我好像也是这个错，我直接改成我博客地址https//zhaoshengmu.top了

**Reply by Chenm4** [reply]
> https://waline.js.org/guide/get-started.html 请对着快速上手操作一遍，不要想当然。serverURL 是指评论服务的地址，不是你的博客地址！
请问一下这个serverURL评论服务的地址指的是哪个？我好像也是这个错，我直接改成我博客地址https//zhaoshengmu.top了

**Reply by lyx-JuneSnow** [reply]
我也遇到了这个问题
```
---
import { SERVER_URL } from "../consts";
---

<br />
<link
  rel="stylesheet"
  href="https://unpkg.com/@waline/client@v3/dist/waline-meta.css"
/>
<div class="comment">
  <div id="waline" data-server-url={SERVER_URL}></div>
</div>
<script type="module">
  import { init } from "https://unpkg.com/@waline/client@v3/dist/waline.js";

  const serverURL = document
    .getElementById("waline")
    .getAttribute("data-server-url");
  init({
    el: "#waline",
    serverURL: 'https://waline-blnx-l5qzavms5-junesnows-projects.vercel.app/',
    dark: 'html[data-theme="dracula"]',
    reaction: [
      "https://saroprock.s3.bitiful.net/good.png",
      "https://saroprock.s3.bitiful.net/bad.png",
      "https://saroprock.s3.bitiful.net/happy.png",
      "https://saroprock.s3.bitiful.net/color.png",
      "https://saroprock.s3.bitiful.net/confuse.png",
      "https://saroprock.s3.bitiful.net/heart.png",
      "https://saroprock.s3.bitiful.net/rocket.png",
      "https://saroprock.s3.bitiful.net/eyes.png",
    ],
    emoji: [
      'https://unpkg.com/@waline/emojis@1.2.0/qq',
      'https://unpkg.com/@waline/emojis@1.2.0/tieba',
    ],
    search: false,
    comment: true,
  });
</script>
```
我的博客地址是https://junesnowblog-junesnows-projects.vercel.app/
可以帮忙解决一下吗感谢

**Reply by HJCheng0602** [reply]
还有一种可能：
把vercel的[Vercel Authentication]设置关掉，我找了半天然后发现我开了这个，一关就好了

---
There is another possibility:
Turn off the [Vercel Authentication] setting of vercel, I searched for a long time and found that I turned on this, and it would be fine for one level

**Comment by yancy-git** [comment]
> 评论时提示Failed to fetch怎么办，已经删了重开了一次还是不行，求救 这是我的博客网址https://changingmoments.vercel.app/

你好，请问解决了吗，我也遇到这个问题了~

**Reply by yancy-git** [reply]
解决了，我是因为密码不对，文档里面说的是邮箱密码，其实要qq邮箱的认证码

**Reply by lizheming** [reply]
文档里没说「是」邮箱密码，文档的原话是「SMTP 邮件发送服务的密码，一般为邮箱登录密码，部分邮箱(例如 163)是单独的 SMTP 密码。」https://waline.js.org/guide/features/notification.html#%E9%82%AE%E4%BB%B6%E9%80%9A%E7%9F%A5

**Reply by zxsking** [reply]
你好我是github登录也会报Failed to fetch这个错，请问你是怎么解决的啊

---
Hello, I log in to github and report the error of Failed to fetch. How did you solve it?

**Reply by lizheming** [reply]
@zxsking serverURL https `https://blogwaline.gking.icu`

**Comment by sunshang-hl** [comment]
![image](https://github.com/user-attachments/assets/024c87d0-214e-404f-af6f-e2dee85bcc42)
serverURL 填Domains下的网址

---
![image](https://github.com/user-attachments/assets/024c87d0-214e-404f-af6f-e2dee85bcc42)
serverURL fill in the URL under Domains

**Reply by owlx56** [reply]
遇到了同样的问题，你的方法是正确的

---
I encountered the same problem, your method is correct

**Reply by 21346asz** [reply]
太感谢了
😭😭😭😭

---
Thank you so much
😭😭😭😭

**Comment by AquariusLeo** [comment]
部署完成后，vercel.app这个域名在国内不挂梯子基本访问不了啊

---
After the deployment is completed, the domain name vercel.app is basically unavailable if it does not hang a ladder in the country

**Reply by SinzMise** [reply]
是这样的，所以才叫你绑定域名

---
That’s the case, that’s why I asked you to bind the domain name
